README.md

```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
+It discusses some member side configuration options to ease understanding Hazelcast's ecosystem. Then, client side configuration options
+regarding cluster connection are discussed. Configuration material regarding data structures are discussed in the following sections.
+You can refer to [IMDG Documentation](https://hazelcast.org/documentation/) and [CONFIG.md](CONFIG.MD) for more information.
+
+## IMDG Configuration
+There are two ways to configure Hazelcast IMDG. One is to use a `hazelcast.xml` file and the other is to programmatically configure the
+instance before starting it from Java code. Since we use standalone servers, we will use `hazelcast.xml` to configure our cluster members.
+
+When you download and unzip hazelcast-<version>.zip, you see the `hazelcast.xml` in /bin folder. When a Hazelcast member starts, it looks for
+`hazelcast.xml` file to load configuration from. In `hazelcast.xml` file, you will see the default configuration. We will go over some important
+elements here:
+
+- `<group>`: This tag specifies to which cluster this member belongs to. A member only connects to other members that are in the same group as
+itself. You will see `<name>` and `<password>` tags with some preconfigured values. You may give your clusters different names so that they can
+live in the same network without disturbing each other. Note that the cluster name should be the same across all members and clients that belong
+ to the same cluster. `<password>` tag is not in use since Hazelcast 3.9. It is there for backward compatibility
+purposes. You can remove or leave it as it is if you use Hazelcast 3.9 or later.
+- `<network>`
+    - `<port>`: This specifies the port number that the member will use when it starts. You can specify a port number. If you set `auto-increment`
+    to `true`, than Hazelcast will try subsequent ports until it finds an available port or `port-count` is reached.
+    - `<join>`: This tag specifies the strategies that member uses to find other members of the cluster. Choose which strategy you want to
+    use by setting its `enabled` attribute to `true` and the others to `false`.
+        - `<multicast>`: Members find each other by sending multicast requests to the specified address and port. It is very useful if ip addresses
+        of the members are not static.
+        - `<tcp>`: This strategy uses a pre-configured list of known members to find and already existing cluster. It is enough for a member to
+        find only one cluster member to connect to the cluster. The rest of the member list is automatically retrieved from that member. We recommend
+        putting multiple known member addresses there to avoid disconnectivity should one of the members in the list is unavailable at the time
+        of connection.
+
+These configuration elements are enough for most connection scenarios. Now we will move onto configuration of the Node.js client.
+
+## Hazelcast Client Configuration
```

Comment:
This part is similar with configuration options in the PR #342 .

Reply:
This part looks like repeated in more than two places. I think redundancy in a guide is not a bad thing though. I say let us keep redundant parts for now. If we find them distracting or completely useless, we can remove them at the end.